src/Engine/GameEvents/CreateUnitEvent.java

```diff
@@ -53,16 +53,11 @@ public void performEvent(MapMaster gameMap)
       myCommander.money -= myNewUnit.model.getCost();
       myCommander.units.add(myNewUnit);
       gameMap.addNewUnit(myNewUnit, myBuildCoords.xCoord, myBuildCoords.yCoord);
+      myCommander.myView.revealFog(myNewUnit);
```

Comment:
Yeah, I noticed that bug, but I couldn't think of a way to fix it in the brief time I spent considering it before laziness and apathy took over.